src/lib.rs

```diff
@@ -683,7 +744,30 @@ pub enum Region<T> {
     /// A possibly empty heap-allocated region, represented as a vector.
     Heap(Vec<T>),
     /// A mmaped region, represented by a vector and its backing memory mapping.
-    MMap(Vec<T>, Option<Mem>),
+    MMap(MMapRegion<T>),
+}
+
+/// Type encapsulating private data for memory-mapped regions.
+pub struct MMapRegion<T> {
+    /// Vector-representation of the underlying memory. Must not be dropped.
+    inner: ManuallyDrop<Vec<T>>,
+    /// The actual memory, so we can recycle it. Option to allow moving.
+    mem: Option<Mem>,
```

Comment:
this can probably just be a bool and you can use https://doc.rust-lang.org/std/vec/struct.Vec.html#method.as_mut_ptr on the `Vec` to recover it whenever?